### PR TITLE
feat(popover): new delay property

### PR DESCRIPTION
**Pull request by nickonometry**
## PR Checklist

Please check if your PR fulfills the following requirements:

- Tests for the changes have been added/updated: [Y]
- Docs have been added/updated: [Y]
- Does this PR introduce a breaking change? [N]
- I have linked any related GitHub issues to be closed when this PR is merged? [Y]

## Describe the new behavior?
When the trigger type of the popover was set to hover, it felt like the popover was appearing too fast. Adding the delay property keeps it consistent with the tooltip, and allows us to control the speed at which the popover element appears

### CODE REVIEW

File: src/lib/tooltip/tooltip.test.ts [hunk 1]
```diff
@@ -434,6 +434,17 @@ describe('Tooltip', () => {
       expect(harness.isOpen).to.be.true;
     });
 
+    it('should open when hovering the trigger button and a delay is set', async () => {
+     const harness = await createFixture({ triggerType: 'hover', delay: 500 });
```
Comment (DRiFTy17) [inline src/lib/tooltip/tooltip.test.ts:438]: Did you intend to add this test to the `popover.test.ts` file instead?

File: src/lib/popover/popover-foundation.ts [hunk 2]
```diff
@@ -304,7 +307,13 @@ export class PopoverFoundation extends BaseClass implements IPopoverFoundation {
       if (!this._persistentHover) {
         this._adapter.addAnchorListener('mouseleave', this._anchorMouseleaveListener);
       }
-      this._openPopover();
+      if (this._delay) {
+        this._hoverTimeout = window.setTimeout(() => {
+        this._openPopover();
+       }, this._delay);
```
Comment (DRiFTy17) [inline src/lib/popover/popover-foundation.ts]: Can you fix the indentation on these two lines?

File: src/lib/popover/popover-foundation.ts [hunk 3]
```diff
@@ -526,6 +536,16 @@ export class PopoverFoundation extends BaseClass implements IPopoverFoundation {
     }
   }
 
+  public get delay(): number {
+    return this._delay;
+  }
+  public set delay(value: number) {
+    if (this._delay !== value) {
+      this._delay = value;
```
Comment (DRiFTy17) [inline src/lib/popover/popover-foundation.ts]: Let's add some safe guards for `NaN` and negative numbers here and just use the default hover delay if detected. May be worth updating the tooltip `delay` property to guard against that the same way as well.

```typescript
public set delay(value: number) {
  if (isNaN(value) || value < 0) {
    value = POPOVER_CONSTANTS.defaults.hoverDelay;
  }
  if (this._delay !== value) {
    // ...
  }
}
```

We should add a test that captures trying to set these values as well.

File: src/dev/pages/popover/popover.html [hunk 4]
```diff
@@ -4,6 +4,7 @@
     title: 'Popover',
     includePath: './pages/popover/popover.ejs',
     options: [
+      { type: 'text-field', inputType: 'number', id: 'opt-delay', label: 'Delay', defaultValue: 0 },
```
Comment (DRiFTy17) [inline src/dev/pages/popover/popover.html:7]: Let's rename the `label` to "Hover delay", and `id` to "opt-hover-delay" for consistency, and update the demo `.ts` file as well.

File: src/lib/tooltip/tooltip.test.ts (hunk 1 above)
Comment (nickonometry) [reply src/lib/tooltip/tooltip.test.ts:438]: I was referencing the tooltip file and accidentally added it there. I can remove it from there and move it over to the popover

File: src/lib/tooltip/tooltip.test.ts [hunk 5]
```diff
@@ -434,6 +434,17 @@ describe('Tooltip', () => {
       expect(harness.isOpen).to.be.true;
     });
 
+    it('should open when hovering the trigger button and a delay is set', async () => {
```
Comment (DRiFTy17) [inline src/lib/tooltip/tooltip.test.ts:437]: You can remove this test since it's a duplicate

File: src/lib/popover/popover.test.ts [hunk 6]
```diff
@@ -1428,6 +1450,7 @@ interface IPopoverFixtureConfig {
   animationType?: PopoverAnimationType;
   triggerType?: PopoverTriggerType;
   persistentHover?: boolean;
+  hoverDelay?: any;
```
Comment (DRiFTy17) [inline src/lib/popover/popover.test.ts:1453]: Use `number` here instead and for your NaN test just set `hoverDelay: 'Testing' as any` to force override the type. We want strong typing for everything else on this property.

File: src/lib/popover/popover.ts [hunk 7]
```diff
@@ -138,6 +142,9 @@ export class PopoverComponent extends OverlayAware<IPopoverFoundation> implement
       case POPOVER_CONSTANTS.attributes.PERSISTENT_HOVER:
         this.persistentHover = coerceBoolean(newValue);
         return;
+      case POPOVER_CONSTANTS.attributes.HOVER_DELAY:
+        this.delay = coerceNumber(newValue);
```
Comment (DRiFTy17) [inline src/lib/popover/popover.ts:146]: You forgot to rename this property

**Review by DRiFTy17 — DISMISSED** [review]

**Review by DRiFTy17 — APPROVED** [review]

**Pull request merged**